aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -498,6 +498,50 @@ Tensor& mean_out(Tensor& result, const Tensor& self, DimnameList dim,
   return at::mean_out(result, self, dimnames_to_positions(self, dim), keepdim, opt_dtype);
 }
 
+Tensor& nanmean_out_cpu_gpu(Tensor& result, const Tensor& self, IntArrayRef dim, bool keepdim, c10::optional<ScalarType> opt_dtype) {
+  ScalarType scalarType = opt_dtype.has_value() > opt_dtype.value() : self.scalar_type();
+  TORCH_CHECK(
+      at::isFloatingType(scalarType) || at::isComplexType(scalarType),
+      "Can only calculate the mean of floating types. Got ",
+      toString(scalarType),
+      " instead.");
+  ScalarType dtype = get_dtype(result, self, opt_dtype, true);
+  Tensor without_nans = torch.where(torch.isnan(self), torch.zeros_like(self), self);
+  if (self.device().is_cpu()) {
+    int64_t dim_prod = at::sum(~self.isnan(), axis=dim, keepdims=keepdim, dtype=dtype);
+    at::sum_out(result, without_nans, dim, keepdim, dtype).div_(dim_prod);
```

Comment:
`.div_` throws a `RuntimeWarning` (and will soon through an error instead, see: https://github.com/pytorch/pytorch/pull/38620) in Python. Let's change this to `true_divide` after a little testing.